### ISSUE TITLE
[Hold for payment 2022-01-27] Add aria-role and aria-label to Terms of Service and Privacy Policy

### ISSUE THREAD

**Issue by isagoico** [Expensify/App#5082]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Navigate to settings 
2. Click on about
3. Using the navigation keys, try to click on Terms of Service or Privacy Policy 

## Expected Result:
User should be able to reach the terms of service and privacy policy by using the navigation keys 

## Actual Result:
Terms of Service & Privacy Policy are non-accessible (non-tabbable, can’t be read & opened by screen reader). 

## Workaround:
Unknown. 

## Platform:

Where is this issue occurring?

- Web
- Desktop App

**Version Number:** 1.0.93-0

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** Any additional supporting documentation

https://user-images.githubusercontent.com/44479856/132074528-5c21bc34-535c-40fd-9586-a25d69b50a2c.mov

**Expensify/Expensify Issue URL:**

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

___
From Mateo Morris (Couldn't find the user in GH) https://expensify.slack.com/archives/C01GTK53T8Q/p1630607495347700

> Terms of Service & Privacy Policy are non-accessible (non-tabbable, can’t be read & opened by screen reader). Just needs an aria-role & aria-label.

**Comment by MelvinBot** [comment]
Triggered auto assignment to @johnmlee101 (`Engineering`), see https://stackoverflow.com/c/expensify/questions/4319 for more details.

**Comment by PrashantMangukiya** [comment]
### Proposed Solution:

We have to set **accessible** and **accessibilityLabel** for terms of service and privacy policy link within src/pages/settings/AboutPage.js

At present accessible and accessibilityLabel not added as shown code below:
https://github.com/Expensify/App/blob/6cd518e81fb60415ba8d4d4a659fdc5dfa95b6fd/src/pages/settings/AboutPage.js#L130-L133

https://github.com/Expensify/App/blob/6cd518e81fb60415ba8d4d4a659fdc5dfa95b6fd/src/pages/settings/AboutPage.js#L143-L146

Once code updated for both it looks as under:
```javascript
  <Text
      style={[styles.chatItemMessageHeaderTimestamp, styles.link]}
      onPress={() => Linking.openURL(CONST.TERMS_URL)}
      accessible
      accessibilityLabel={translate('initialSettingsPage.readTheTermsAndPrivacyPolicy.phrase2')}
  >
```

```javascript
  <Text
      style={[styles.chatItemMessageHeaderTimestamp, styles.link]}
      onPress={() => Linking.openURL(CONST.PRIVACY_URL)}
      accessible
      accessibilityLabel={translate('initialSettingsPage.readTheTermsAndPrivacyPolicy.phrase4')}
  >
```

At present External label not set with this issue, so just giving suggestion. It might be helpful for someone if handled internally. 

If this issue consider as External and my solution accepted then kindly invite me on Upwork project. Here is my Upwork profile: https://www.upwork.com/freelancers/~01c718889ed7821f82

Thank you.

**Comment by mateomorris** [comment]
**Proposal**

After taking another look, it turns out there's an existing component we can use instead which includes the aria-role. And since the link text is descriptive, `aria-label` is redundant: "The aria-label attribute provides a way to place a descriptive text label on an object, such as a link, when there are no elements visible on the page that describe the object" [w3.org](https://www.w3.org/WAI/WCAG21/Techniques/aria/ARIA8)

So these lines here 
https://github.com/Expensify/App/blob/6cd518e81fb60415ba8d4d4a659fdc5dfa95b6fd/src/pages/settings/AboutPage.js#L130-L150

should be replaced with
```
<TextLink
    style={[styles.chatItemMessageHeaderTimestamp, styles.link]}
    href={CONST.TERMS_URL}
>
    {translate(
        'initialSettingsPage.readTheTermsAndPrivacyPolicy.phrase2',
    )}
</TextLink>
{' '}
{translate(
    'initialSettingsPage.readTheTermsAndPrivacyPolicy.phrase3',
)}
{' '}
<TextLink
    style={[styles.chatItemMessageHeaderTimestamp, styles.link]}
    href={CONST.PRIVACY_URL}
>
    {translate(
        'initialSettingsPage.readTheTermsAndPrivacyPolicy.phrase4',
    )}
</TextLink>
```

https://user-images.githubusercontent.com/39444813/132089833-3258330b-be47-4a5a-a5fd-a65e47df7fdb.mov

**Comment by PrashantMangukiya** [comment]
Hi @mateomorris , your solution also looks good. One thing I notice is that whenever we use TextLink component as per your suggestion it shows  pipe at left and right of text. If we add accessibilityLabel to Text component as per my solution, it shows box around the text, i.e. similar to shows box around close and back button on that page when tabbing. i.e. Both approach have same result but some visual difference, not sure it will make any difference from usage/ui point of view or not. Thanks.

**Comment by mateomorris** [comment]
@PrashantMangukiya I noticed that, but in this case it's still better to use TextLink because it renders to an anchor tag as opposed to a span (i.e. it's accessible to both keyboard and screenreader users). A compromise would be to add `accessibilityRole="link"`, but accessibility-wise it's always preferable to use native web elements whenever possible.

**Comment by MelvinBot** [comment]
@johnmlee101 Whoops! This issue is 2 days overdue. Let's get this updated quick!

**Comment by johnmlee101** [comment]
Hey all, love the proactiveness here. Going to mark as external and review the two proposals.

**Comment by johnmlee101** [comment]
I'm leaning towards @mateomorris 's implementation since TextLink already has the accessibilityRole defined as part of the component, so we don't have to add new attributes to do the same thing. However, the only downside is the weird piping highlighting, but I think that all TextLinks deal with that today? So it seems like a larger issue that should be addressed with that component, rather than the issue here.

**Comment by johnmlee101** [comment]
Both implementations are great, and I appreciate you both coming up with solutions early on!

**Comment by JmillsExpensify** [comment]
@mateomorris Upwork job posted here! https://www.upwork.com/jobs/~015732100d25e1724d

**Comment by Beamanator** [comment]
Just creeping on some issues 😅 If we're assigning the job to @mateomorris let's assign this issue to him 👍

**Comment by JmillsExpensify** [comment]
@johnmlee101 Mind confirming?

**Comment by johnmlee101** [comment]
Updated the assignment!

**Comment by kadiealexander** [comment]
Hi @mateomorris, I've placed this issue on hold as per [this update](https://expensify.slack.com/archives/C01GTK53T8Q/p1631754482374800), as we are prioritising issues related to a feature release scheduled for 9/24. As an apology for the delay, we will add a $100 bonus as a thank you for waiting.

**Comment by JmillsExpensify** [comment]
Holding per Kadie's message

**Comment by JmillsExpensify** [comment]
Still holding per Kadie's message

**Comment by JmillsExpensify** [comment]
Still holding

**Comment by MelvinBot** [comment]
@JmillsExpensify, @johnmlee101, @mateomorris Uh oh! This issue is overdue by 2 days. Don't forget to update your issues!

**Comment by johnmlee101** [comment]
Still on hold!

**Comment by kadiealexander** [comment]
Please refer to [this post](https://expensify.slack.com/archives/C01GTK53T8Q/p1633557259458600) for updated information on the `n6 hold`, we've raised the bonus to **$250** for all issues where a **PR is created** before the N6 hold is lifted.

**Comment by MelvinBot** [comment]
This issue has not been updated in over ***15*** days. @JmillsExpensify, @johnmlee101, @mateomorris eroding to ***Monthly*** issue.

P.S. Is everyone reading this sure this is really a near-term priority? Be brave: if you disagree, go ahead and close it out. If someone disagrees, they'll reopen it, and if they don't: one less thing to do!

**Comment by Beamanator** [comment]
The `n6-hold` ended a few weeks ago so I believe we can merge https://github.com/Expensify/App/pull/5173 soon

**Comment by johnmlee101** [comment]
Bumped the PR

**Comment by JmillsExpensify** [comment]
Looks like we're waiting on an approval in the linked PR. Staying the course until then!

**Comment by JmillsExpensify** [comment]
Waiting two more days for no regressions and then I'll process the payment.

**Comment by JmillsExpensify** [comment]
@mateomorris Hmm, I wasn't able to find an active contract in Upwork for this PR. Probably a mistake on my part. Have you been paid for this job yet?

**Comment by mateomorris** [comment]
@JmillsExpensify strange, it must have expired since it was from a while ago. I haven't been paid for it.

**Comment by JmillsExpensify** [comment]
Strange indeed. I've gone ahead and created another active job for your work: https://www.upwork.com/jobs/~010c60fca196d9f4ad. Please apply here and I'll make sure this is processed for you. Thanks!

**Comment by JmillsExpensify** [comment]
@mateomorris Just checking in. I'm ready to process the payment and close this issue when you're able to apply in Upwork. Thank you!

**Comment by mateomorris** [comment]
Thanks for the reminder @JmillsExpensify, just applied. I increased the proposal amount to account for the Hold (250) and for discovering the issue (250, don't think that was paid out before).

**Comment by JmillsExpensify** [comment]
All checked out! Offer sent, will process as soon you accept and then we can close this issue. 😄

**Comment by JmillsExpensify** [comment]
Payment processed and contract closed. Thank you!
